### ISSUE TITLE
Limit data precision in cleanfigure

### ISSUE THREAD

**Issue by egeerardyn** [matlab2tikz/matlab2tikz#757]
This has been requested before in https://github.com/matlab2tikz/matlab2tikz/pull/695#issuecomment-134517983 :

> For the numerical issues, I would implement an optional filter in cleanfigure that takes an alpha such that the precision of all data points is reduced to EPS = alpha * eps(V), where V is some measure of the maximum X/Y/Z value/range in an axis (could either be from the limits or just from the data itself).
> I imagine something like: round(values / EPS) * EPS
> So if you have a Y = sin(X), eps will be around 5e-16, so that should put 0 instead of those 1....e-16 values.

Basically, the problem is that in some plots we might have `1e-16` together with values likes `10`. Obviously, `1e-16` will be displayed at the location `0`, so it would be beneficial to actually do this already on our end as that
 - is clearer,
 - takes less space (= less likely to overflow any LaTeX buffer),
 - less likely causes our hash-based test suite to fail.

Since this would alter the data, the best place for this is in `cleanfigure` (i.e. `matlab2tikz` shouldn't change the user's data).

The implementation should be parametrized for the required (relative?) numerical precision in some useful way (e.g. as with `alpha` in the quote above) such that a user can pick the numerical precision he wants.

**Note**: slightly related, there are occasions where actually going the other way (i.e. dithering a `0` to `1e-16`) would be beneficial, e.g. #318 and maybe some other plots with log-scaled axes?

**Comment by miscco** [comment]
So one way to address this problem would be the usage of relative data coordinates.

scale = max(abs(data), [], 1);
data_scaled  = bsxfun(@rdivide, data, scale);

Then operate on data_scaled and afterwards

data  = bsxfun(@times, data_scaled, scale); 

This would have the advantage, that it should in principle be independent of scale. We could even set a fixed precision to handle data_scaled

**Comment by egeerardyn** [comment]
That's indeed a good approach.
My first idea would be to use the `eps(scale)` value (since I'm accustomed to that), but any way to tackle this is fine as long as it compensates for the nominal scale of the data.

The most tricky part of this feature, I guess, is dealing with possible edge cases (e.g. when the data contains `NaN` of `Inf`).

**Comment by miscco** [comment]
So you would propose something along the lines of 
```
scale = max(abs(data), [], 1);
data_scaled  = bsxfun(@rdivide, data, scale);
data  = bsxfun(@times, round(data), scale);
```

Regarding NaN and Inf there should be no problem, as round, times and divide always return NaN and Inf respectively

**Comment by miscco** [comment]
Btw should it be 
```matlab
scale = 2*max(abs(data,[],1));
```
As the dta range would be from [-scale, scale]?

**Comment by miscco** [comment]
As this has been merged #791  i think we can close that issue.
